sdks/java/io/csv/src/test/java/org/apache/beam/sdk/io/csv/CsvIOParseHelpersTest.java

```diff
@@ -33,6 +34,77 @@
 @RunWith(JUnit4.class)
 public class CsvIOParseHelpersTest {
 
+  /** Tests for {@link CsvIOParseHelpers#validate(CSVFormat)}. */
+  @Test
+  public void givenCSVFormatWithHeader_validates() {
+    CSVFormat format = csvFormatWithHeader();
+    CsvIOParseHelpers.validate(format);
+  }
+
+  @Test
+  public void givenCSVFormatWithNullHeader_throwsException() {
+    CSVFormat format = csvFormat();
+    String gotMessage =
+        assertThrows(IllegalArgumentException.class, () -> CsvIOParseHelpers.validate(format))
+            .getMessage();
+    assertEquals("Illegal class org.apache.commons.csv.CSVFormat: header is required", gotMessage);
```

Comment:
It's good to check error message. In general, there is no guarantee that upstream error message will keep the same for version upgrade.

A less strict check is assertThat(gotMessage, contains("header is required"))

Reply:
I like that we are asserting the error message contents so that we can validate we are sending the correct message to the developer.

Reply:
I also think it's important to check the error message because each of the methods in `CsvIOParseHelpers::validate(CSVFormat)` throws the same `IllegalArgumentException`, so the error message will ensure that the right `IllegalArgumentException` exception is thrown given a particular test condition.

Reply:
IllegalArgumentException is already checked by "assertThrows". The Exception type is generally more stable than the error message.